### DISCUSSION TITLE
Declare type for relational fields

### DISCUSSION THREAD

**Discussion by frozberry** [prisma/prisma#12042] [Q&A] [answered]
What's the idiomatic way of declaring the type of an relational field from the autogenerated TS types?

Let's say I have a simple Schema of `User` and `Post`:
```prisma
model User {
  id    String @id @default(uuid())
  name  String
  posts Post[]
}

model Post {
  id      String @id @default(uuid())
  content String
  user    User   @relation(fields: [userId], references: [id])
  userId  String @unique
}
```

I want to create a function that loops over and prints all a user's `Post` contents:
```ts
const print = (user) => {
  user.posts.forEach((post) => {
    console.log(post.title)
  })
}
```

I'm using Typescript, so I import the auto generated `User` type:

```ts
import { User } from "@prisma/client"

const print = (user: User) => {
  user.posts.forEach((post) => {
    console.log(post.title)
  })
}
```

But the `posts` field is missing from the auto generated `User` type.

I can work around it by manually declaring the type:
```ts
import { User, Post } from "@prisma/client"

type UserWithPost = User & { posts: Post[] }

const print = (user: UserWithPost) => {
  user.posts.map((post) => {
    console.log(post.title)
  })
}
```

But that's not very elegant code, and and becomes clunky as the code becomes more complex. What's the idiomatic way of utlising the `select` fields with prisma and Typescript?

**Comment by austincrim** [comment, marked as answer]
Hey @frozberry 👋, 

Since Prisma queries do not include relations by default (you have to use the include option), the generated types do not include them either. You can create the type you are looking for using one of our built-in utility types, though.
```ts
import { Prisma } from '@prisma/client'
type UserWithPost = Prisma.UserGetPayload<{
  include: { post: true }
}>
```
Here we use the `Prisma.UserGetPayload` type and pass in query options as the generic parameter to get the desired type.

Let me know if I can clarify anything!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
